### DISCUSSION TITLE
非常舒服的输入法，希望能够支持小鹤音形

### DISCUSSION THREAD

**Discussion by Attiv** [Cantoboard/Cantoboard#62] [General]
UI 好看，还流畅，使用起来很舒服。希望能够支持小鹤音形

**Comment by alex-the-man** [comment]
Are you able to build the keyboard extension? I could show you how to replace the dict.

**Reply by Attiv** [reply]
Yes, thanks. I have used my rime config files instead of the original rime config and rebuilt the project. But it seems that my config file is not working.

**Reply by Attiv** [reply]
Hi @alex-the-man, when will u have time to help me with how to replace the dict?

**Reply by alex-the-man** [reply]
Could you place the dict.yaml under this directory, then change default.yaml to load the scheme?
https://github.com/Cantoboard/Cantoboard/tree/main/CantoboardFramework/Data/Rime

After that, change/add the new schema to this class. 

https://github.com/Cantoboard/Cantoboard/blob/2c2d74aa47b640d773caa74e9945f18b4475224e/CantoboardFramework/Keyboard/InputEngine/RimeInputEngine.swift#L13

If you are adding a new schema, you might want to add it to the menu as well:

https://github.com/Cantoboard/Cantoboard/blob/2c2d74aa47b640d773caa74e9945f18b4475224e/CantoboardFramework/Keyboard/View/Status/StatusMenuHandler.swift#L70

Then, run CantoboardTestApp in the simulator to generate the Rime dictionaries and prisms. Also, test your input method there.

When CantoboardTestApp's starting, look at the log in XCode. It will show where the container folder is (userDataPath).

https://github.com/Cantoboard/Cantoboard/blob/2c2d74aa47b640d773caa74e9945f18b4475224e/CantoboardFramework/Keyboard/InputEngine/RimeApi.swift#L87

Open the folder in Finder, copy all the generated dicts to this folder: https://github.com/Cantoboard/Cantoboard/tree/main/CantoboardFramework/Data/Rime/build

Then, build and run Cantoboard and test the input method.

**Reply by Attiv** [reply]
@alex-the-man Thanks a lot, I have replaced `luna_pinyin` in `RimeinputEngine.swift` yesterday, and it worked. Now I am going to debug how to auto input with 4 key typed
